pkg/basicstation/cups/config.go

```diff
@@ -61,7 +61,7 @@ func (conf ServerConfig) NewServer(c *component.Component, customOpts ...Option)
 			}),
 		)
 	}
-	if tlsConfig, err := c.GetTLSServerConfig(c.Context()); err == nil {
+	if tlsConfig, err := c.GetTLSClientConfig(c.Context()); err == nil {
```

Comment:
This is very confusing when passing it to `NewServer`, so it could benefit from a little code comment.